src/ethereum/EthereumVerifier.js

```diff
@@ -365,11 +365,11 @@ class EthereumVerifier {
       // Unique hash for epoch, secret contract address, and nonce
       const hash = cryptography.hash(abi.rawEncode(
           ['uint256', 'bytes32', 'uint256'],
-          [params.seed, secretContractAddress, nonce]
+          [params.seed, nodeUtils.add0x(secretContractAddress), nonce]
         ));
 
       // Find random number between [0, tokenCpt)
-      let randVal = JSBI.remainder(cryptography.toBN(hash), cryptography.toBN(tokenCpt));
+      let randVal = JSBI.remainder(cryptography.toBN(hash), tokenCpt);
```

Comment:
Why did you remove the conversion of the `tokenCpt` to big number?

Reply:
Because it already becomes a Big Number in the array reduction step a couple lines above.